### PR TITLE
Allow setting layers on a root view with forks

### PR DESCRIPTION
**Pull request by invisig0th**

### CODE REVIEW

File: synapse/lib/view.py [hunk 1]
```diff
@@ -777,6 +775,34 @@ async def setLayers(self, layers):
...
+            byparent[view.parent].append(view)
+
+        while todo:
+
+            view = todo.pop()
+
+            for child in byparent.get(view, ()):
+
+                layers = list(view.layers)
+                layers.append(child.layers[-1])
```
Comment (MichaelSquires) [inline synapse/lib/view.py:799]: Isn't `child.layers[-1]` the first layer in the bottom view. Do you want the top layer of the view? That should be `child.layers[0]`, right?

Comment (MichaelSquires) [reply synapse/lib/view.py:799]: Check out the layer ordering in this screenshot. The first print is from `test_lib_stormtypes.py:3470` and the second is from `test_lib_stormtypes.py:3476`.
image

Notice the top and bottom layers are identical after adding the new layer.

File: synapse/tests/test_lib_stormtypes.py [hunk 2]
```diff
@@ -3465,10 +3465,16 @@ async def test_storm_lib_view(self):
...
+            # fork the default view to test editing the root view layers
+            fork00 = await core.callStorm('return($lib.view.get().fork().iden)')
+            self.eq(2, await core.callStorm('return($lib.view.get().layers.size())', opts={'view': fork00}))
+
             await core.callStorm('$lib.view.get().set(layers, $layers)', opts={'vars': {'layers': layrs0}})
             ldefs = await core.callStorm('return($lib.view.get().get(layers))')
             self.eq(layrs0, [x.get('iden') for x in ldefs])
 
+            self.eq(3, await core.callStorm('return($lib.view.get().layers.size())', opts={'view': fork00}))
+
```
Comment (MichaelSquires) [inline synapse/tests/test_lib_stormtypes.py:3477]: Check layer idens here to make sure the new fork has the expected layers and in the right order.

**Review by MichaelSquires — APPROVED** [review]
good catch on the layer ordering in your last commit!

File: synapse/lib/view.py [hunk 3]
```diff
@@ -777,6 +775,34 @@ async def setLayers(self, layers):
...
+
+            view = todo.pop()
+
+            for child in byparent.get(view, ()):
+
+                layers = [child.layers[0]]
+                layers.extend(view.layers)
+
+                child.layers = layers
+                await child.info.set('layers', [layr.iden for layr in layers])
```
Comment (vEpiphyte) [inline synapse/lib/view.py:802]: I believe that this would need beholder events for each modified view.

**Review by Cisphyx — APPROVED** [review]

**Pull request merged**